addons/web_editor/static/tests/html_field_tests.js

```diff
@@ -36,13 +124,199 @@ QUnit.module("WebEditor.HtmlField", ({ beforeEach }) => {
+                </form>`,
+        });
+        await click(target, '.o_form_button_save');
+        assert.containsOnce(target, ".o_notification");
+        const notif = target.querySelector(".o_notification");
+        assert.strictEqual(
+            notif.querySelector(".o_notification_content").textContent,
+            "Header"
+        );
+        assert.hasClass(notif, "border-danger");
+
+    });
+
+    QUnit.test('colorpicker', async function (assert) {
+        assert.expect(10);
+
+        serverData.models['note.note'].records = [{
+            id: 1,
+            display_name: "first record",
+            header: "<p>  &nbsp;&nbsp;  <br>   </p>",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
+
+        // Add the ajax service (legacy), because wysiwyg RPCs use it.
+        patchWithCleanup(legacyEnv, {
+            services: {
+                ...legacyEnv.services,
+                ajax: {
+                    rpc: async (route, args) => {
+                        if (args.model === "ir.ui.view" && args.method === 'render_public_asset') {
+                            if (args.args[0] === "web_editor.colorpicker") {
+                                return COLOR_PICKER_TEMPLATE;
+                            }
+                            if (args.args[0] === "web_editor.snippets") {
+                                return SNIPPETS_TEMPLATE;
+                            }
+                        }
+                    },
+                },
+            }
+        });
+
+        await makeView({
+            type: "form",
+            resId: 1,
+            resModel: "note.note",
+            serverData,
+            arch: `
+                <form>
+                    <field name="body" widget="html" style="height: 100px"/>
+                </form>`,
+        });
+
+        const field = target.querySelector('.o_field_html[name="body"]');
+        const editable = field.querySelector("[contenteditable='true']");
+        await click(editable);
+
+        assert.strictEqual(
+            window.getComputedStyle(document.querySelector("#toolbar")).getPropertyValue("visibility"),
+            "hidden",
+            "should hide the toolbar");
+
+        // select the text
+        const pText = editable.querySelector("p").firstChild
+        Wysiwyg.setRange(pText, 1, pText, 10);
+        // text is selected
+
+        var range = Wysiwyg.getRange();
+
+        assert.strictEqual(range.sc, pText,
+            "should select the text");
+
+        await wait(1);
+        assert.strictEqual(
+            window.getComputedStyle(document.querySelector("#toolbar")).getPropertyValue("visibility"),
+            "visible",
+            "should show the toolbar");
+
+
+        assert.ok(!document.querySelector(".colorpicker-menu.show"),
+            "should hide the color picker");
+        await click(document.querySelector("#oe-fore-color"), undefined, true);
```

Comment:
What are those parameters doing ? The `undefined` and `true` ? I'm curious.

Reply:
To disable visibility check (bc, qunit-fixture is outside the screen)

Reply:
But then literally every qunit test would need to set those at true, right ? They don't seem to do so.

Reply:
I don't know but without them they don't work and it seems linked to the position of the object or offset. I fixed it using this parameter. I can analyze why the others work. And also the line that deals with out of the screen elements (one sec)